android-lib/src/androidTest/java/com/kinvey/androidTest/store/DeltaSetNewTest.java

```diff
@@ -0,0 +1,766 @@
+        assertEquals(2, testManager.sync(store, usernameQuery).getResult().getCount());
+        person.setAge("20");
+        testManager.save(store, person);
+        assertEquals(1, testManager.sync(store, usernameQuery).getResult().getCount());
+    }
+
+    @Test
+    public void testSyncStoreTypeCache_DisablingDeltaSet() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.CACHE);
+        testManager.save(store, new Person(TEST_USERNAME));
+        assertEquals(1, testManager.sync(store, emptyQuery).getResult().getCount());
+        assertEquals(0, testManager.sync(store, emptyQuery).getResult().getCount());
+        store.setDeltaSetCachingEnabled(false);
+        assertEquals(1, testManager.sync(store, emptyQuery).getResult().getCount());
+    }
+
+    @Test
+    public void testFindStoreTypeCache_CreateItem() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.CACHE);
+        testManager.save(store, new Person(TEST_USERNAME));
+        assertEquals(1, testManager.find(store, emptyQuery).getResult().getResult().size());
+        testManager.save(store, new Person(TEST_USERNAME_2));
+        assertEquals(2, testManager.find(store, emptyQuery).getResult().getResult().size());
+    }
+
+    @Test
+    public void testFindStoreTypeCache_UpdateItem() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.CACHE);
+        Person person = testManager.save(store, new Person(TEST_USERNAME)).getResult();
+        assertEquals(1, testManager.find(store, emptyQuery).getResult().getResult().size());
+        person.setAge("20");
+        testManager.save(store, person);
+        assertEquals(1, testManager.find(store, emptyQuery).getResult().getResult().size());
+    }
+
+    @Test
+    public void testFindStoreTypeCache_UpdateAndDeleteItem() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.CACHE);
+        Person person = testManager.save(store, new Person(TEST_USERNAME)).getResult();
+        assertEquals(1, testManager.find(store, emptyQuery).getResult().getResult().size());
+        person.setAge("20");
+        testManager.save(store, person);
+        assertEquals(1, testManager.find(store, emptyQuery).getResult().getResult().size());
+        testManager.delete(store, person.getId());
+        assertEquals(0, testManager.find(store, emptyQuery).getResult().getResult().size());
+    }
+
+    @Test
+    public void testFindStoreTypeCache_DeleteOneOfThreeItems() throws InterruptedException {
+        initDeltaSetCachedCollection(StoreType.CACHE);
+        Person person = testManager.save(store, new Person(TEST_USERNAME)).getResult();
+        testManager.save(store, new Person(TEST_USERNAME));
+        testManager.save(store, new Person(TEST_USERNAME_2));
+        assertEquals(2, testManager.find(store, usernameQuery).getResult().getResult().size());
+        testManager.delete(store, person.getId());
```

Comment:
This test is supposed to have 3 items in the store, but only 2 are asserted. Why is this?